sympy/series/formal.py

```diff
@@ -1143,6 +1143,66 @@ def integrate(self, x=None, **kwargs):
 
         return self.func(f, self.x, self.x0, self.dir, (ak, self.xk, ind))
 
+    def convolve(self, other, x=None, order=4):
+        """ Convolute two Formal Power Series and return the truncated terms upto specified order.
+
+        Parameters
+        ==========
+
+        order : Number, optional
+            Specifies the order of the term upto which the polynomial should
+            be truncated.
+
+        Examples
+        ========
+
+        >>> from sympy import fps, sin, exp, convolution
+        >>> from sympy.abc import x
+        >>> f1 = fps(sin(x))
+        >>> f2 = fps(exp(x))
+
+        >>> f1.convolve(f2, x, 6)
+        x + x**2 + x**3/3 - x**5/12 + O(x**6)
+
+        See Also
+        ========
+
+        sympy.discrete.convolutions
+        """
+        from sympy.discrete.convolutions import convolution
+
+        if x is None:
+            x = self.x
+        if order is None:
+            return iter(self)
+
+        other = sympify(other)
+        if isinstance(other, FormalPowerSeries):
+            if self.dir != other.dir:
+                raise ValueError("Both series should be calculated from the"
+                                 " same direction.")
+            elif self.x0 != other.x0:
+                raise ValueError("Both series should be calculated about the"
+                                 " same point.")
+
+            trunc = (order + 2)/2 if (order % 2 == 0) else (order + 1)/2
+            coeff1, coeff2, terms = [], [], []
+            for pt in range(int(trunc)):
```

Comment:
`trunc` is already `int` as it appears from L1188

Reply:
Don't think so, if `order ` is Python int ( which it appears to be), the result of division would be a `float` value.

Reply:
Yes, for that `//` should be used rather than `/`. I forgot to mention that.  `//` is compatible both with python and python3.

Reply:
Well, it's just division by 2, bit shifting would have been sufficient. Isn't it? Though don't know if order is allowed to be symbolic.

Reply:
The casting is working similarly. I think it will cover all the cases ( symbolic as well).

Reply:
No issues, that was just a comment. int(symbol) doesn't work though afaik. but it's ok. resolve this conversation.

Reply:
```
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import Symbol
>>> x = Symbol('x')
>>> int(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gagandeep/sympy/sympy/core/expr.py", line 289, in __int__
    raise TypeError("can't convert symbols to int")
TypeError: can't convert symbols to int

```
I am not much familiar with this module so don't know whether the API allows symbolic parameters. If it allows then this function needs a check for it. Otherwise, consider trying out something like,
```
trunc = (order + 2)>>1 if not order&1 else (order + 1)>>1
```
order can never be float as per the definition of fps and afaik.
That's all. :)

Reply:
> That's all. :)

I don't think we are supposed to use `Symbol` as it is. It doesn't convey anything and hence `int()` won't work. But `int()` works for `sympy`'s numeric types like `S(1)` and all. But your solution is also a possible solution.